ksp_plugin_adapter/window_renderer.cs

```diff
@@ -124,6 +124,25 @@ internal abstract class BaseWindowRenderer : ScalingRenderer, IConfigNode {
     }
     UnityEngine.GUI.skin = skin_;
     if (show_) {
+      // N.B.(al2me6): Empirical observation gave the following conclusions:
+      // In each frame, |OnGUI| (from whence this method is ultimately called)
+      // will be called at least twice, with the following order of event types:
+      // 1. Layout
+      // 2. For each interaction during that frame (e.g., mouse event):
+      //   a. Interaction event
+      //   b. Layout
+      // 3. Repaint
+      // cf. <https://docs.unity3d.com/ScriptReference/Event.html>.
+      // Furthermore, |Shrink| is not safe to call in the (time) window between
+      // the call to |GUILayout.Window| in the last Layout of the frame and the
+      // subsequent Repaint; doing so may cause the window to become blank for
+      // one frame.  Thus, here (in a Layout, before drawing the window) is a
+      // safe place to execute any requested shrinks.
+      if (shrink_scheduled_
+          && UnityEngine.Event.current.type == UnityEngine.EventType.Layout) {
+        Shrink();
+        shrink_scheduled_ = false;
```

Comment:
Hm, I intended for `Shrink` to be a private thing called here and nowhere else. In which case, what if I just inlined it here?

Reply:
I'd prefer to keep a clear encapsulation for the-actions-that-perform-shrinking, so I'd vote for keeping `Shrink`.  The precondition of `Shrink` is "shrinking has been scheduled" (we could assert that `shrink_scheduled` is true at the beginning of `Shrink`, but that's probably overly pedantic), the postcondition of `Shrink` is "shrinking has been done and is no longer scheduled", so the boolean should be reset in that function.

Technically, even if `Shrink` is private, it might end up being called from several places in the future (software is like that, we can't anticipate what will happen).  It's a good idea to give it a clear contract, even if there is only a call site at the moment.

Sorry about the nitpicking, this is the kind of stuff @eggrobin and I argue to death in code reviews (often face-to-face).